lib/uri.ml

```diff
@@ -870,7 +870,9 @@ let resolve schem base uri =
     | None, Some _ ->
       {uri with scheme=base.scheme; path=remove_dot_segments uri.path}
     | None, None ->
-      let uri = {uri with scheme=base.scheme; host=base.host; port=base.port} in
+      let uri =
+          let userinfo = if uri.userinfo == None then base.userinfo else uri.userinfo in
```

Comment:
`uri.userinfo == None` should be `uri.userinfo = None`. Even better, the conditional could be `match uri.userinfo with None -> base.userinfo | Some _ -> uri.userinfo`. Finally, definition of `userinfo` doesn't need to be inside of `let uri =` as it's a bit awkward there.

Reply:
@dsheets ,  thank you for looking at this. I believe  c2a281b should resolve concerns.